plugins/pdr_deterministic_plugin/tests/simulation_telemetry.py

```diff
@@ -46,38 +75,69 @@ def do_GET(self):
     RCV_PACKETS_COUNTER:"10000000",
 }
 
-ALL_DATA_TEST = {
-    # all positive tests
-    # iteration, row index, counter name = value
-    (1,0,PHY_SYMBOL_ERROR):0, # example, also negative test
-    (1,3,RCV_ERRORS_COUNTER):50,
+# All positive tests (tested ports should be isolated)
+# (iteration, row index, counter name): value
+POSITIVE_DATA_TEST = {
+    (1, 3, RCV_ERRORS_COUNTER): 50,
     # testing packet drop rate criteria
-    (2,3,RCV_ERRORS_COUNTER):500,
+    (2, 3, RCV_ERRORS_COUNTER): 500,
 
     # testing temperature changes
-    (3,4,TEMP_COUNTER):90,
+    (3, 4, TEMP_COUNTER): 90,
     # testing temperature max difference
-    (3,6,TEMP_COUNTER):25,
+    (3, 6, TEMP_COUNTER): 25,
 
-    (4,8,RCV_REMOTE_PHY_ERROR_COUNTER):50,
+    (4, 8, RCV_REMOTE_PHY_ERROR_COUNTER): 50,
     # testing packet drop rate criteria from the second counter. because we look on rate
-    (5,8,RCV_REMOTE_PHY_ERROR_COUNTER):500,
-    
-    # testing link down
-    (4,2,LINK_DOWN_COUNTER):2,
-    (5,2,LINK_DOWN_COUNTER):3,
-    (6,2,LINK_DOWN_COUNTER):4,
+    (5, 8, RCV_REMOTE_PHY_ERROR_COUNTER): 500,
 
-    # negative tests
-    # testing ber calculation ( should not pass as not all are not equal to 0)
+    # testing link down
+    (4, 2, LINK_DOWN_COUNTER): 2,
+    (5, 2, LINK_DOWN_COUNTER): 3,
+    (6, 2, LINK_DOWN_COUNTER): 4,
+
+    # testing auto remove port from blacklist
+    (0, 9, EXCLUDE_PORT_SHORT_TIME): 60, # add to blacklist for 60 seconds
+    (8, 9, LINK_DOWN_COUNTER): 1,        # at this moment the port should be already automatically removed from blacklist
+    (9, 9, LINK_DOWN_COUNTER): 2,        # try trigger isolation issue
+    # testing forced remove port from blacklist
+    (0, 1, EXCLUDE_PORT_LONG_TIME): 0,   # add to blacklist forever
+    (1, 1, INCLUDE_PORT): -1,            # remove port from blacklist
+    (2, 1, LINK_DOWN_COUNTER): 1,        # at this moment the port should be already removed from blacklist
+    (3, 1, LINK_DOWN_COUNTER): 2,        # try trigger isolation issue
+
+    # testing ber calculation (should not pass as not all are not equal to 0)
+}
 
+# All negaitive tests (tested ports should not be isolated)
+# (iteration, row index, counter name): value
+NEGATIVE_DATA_TEST = {
+    # example, also negative test
+    (1, 0, PHY_SYMBOL_ERROR): 0,
+    
+    # testing blacklist
+    (0, 5, EXCLUDE_PORT_LONG_TIME): 0, # add to blacklist forever
+    (1, 5, LINK_DOWN_COUNTER): 1,
+    (2, 5, LINK_DOWN_COUNTER): 2,      # try trigger isolation issue (should be ignored)
```

Comment:
the comments are not needed, you wrote above that it blacklist test

Reply:
Different blacklist tests consist of several consecutive steps/operations. The steps are different for different blacklist tests. The comments describe what we are expecting from each step. So, I would leave the comments.

Reply:
I will :)